src/input-schemas.json

```diff
@@ -0,0 +1,1028 @@
+      "UoM": null,
+      "constraints": {
+        "OneOf": [
+          "none",
+          "simple",
+          "compact"
+        ]
+      },
+      "default": "none",
+      "description": "How investments are treated.",
+      "type": "VARCHAR"
+    },
+    "is_seasonal": {
+      "UoM": null,
+      "constraints": null,
+      "default": false,
+      "description": "Whether seasonal storage (e.g., hydro) or not (e.g., battery)",
+      "type": "BOOLEAN"
+    },
+    "max_ramp_down": {
+      "UoM": "p.u./h",
+      "constraints": {
+        "minimum": 0
+      },
+      "deafult": 0,
+      "description": "Maximum ramping down rate as a portion of the capacity of asset.",
+      "type": "DOUBLE"
+    },
+    "max_ramp_up": {
+      "UoM": "p.u./h",
+      "constraints": {
+        "minimum": 0
+      },
+      "deafult": 0,
+      "description": "Maximum ramping up rate as a portion of the capacity of asset.",
+      "type": "DOUBLE"
+    },
+    "min_operating_point": {
+      "UoM": "p.u.",
+      "constraints": {
+        "maximum": 1,
+        "minimum": 0
+      },
+      "default": 0,
+      "description": "Minimum operating point or minimum stable generation level defined as a portion of the capacity of asset.",
+      "type": "DOUBLE"
+    },
+    "ramping": {
+      "UoM": null,
+      "constraints": null,
+      "default": false,
+      "description": "Whether asset has ramping constraints or not.",
+      "type": "BOOLEAN"
+    },
+    "storage_method_energy": {
+      "UoM": null,
+      "constraints": null,
+      "default": false,
+      "description": "Whether there is independent investment on storage capacity or not. If false, the investment on storage capacity uses the energy_to_power_ratio as a constant fixed value.",
+      "type": "BOOLEAN"
+    },
+    "technical_lifetime": {
+      "UoM": "years",
+      "constraints": {
+        "minimum": 0
+      },
+      "default": 1,
+      "description": "Technical lifetime of the unit to determine for how long the capacity is considered from the commission year.",
+      "type": "INTEGER"
+    },
+    "type": {
+      "UoM": null,
+      "constraints": {
+        "OneOf": [
+          "producer",
+          "consumer",
+          "storage",
+          "conversion",
+          "hub"
+        ]
+      },
+      "deafult": null,
+      "description": "Type of energy asset.",
+      "type": "VARCHAR"
+    },
+    "unit_commitment": {
+      "UoM": null,
+      "constraints": null,
+      "default": false,
+      "description": "Whether asset has unit commitment constraints or not",
+      "type": "BOOLEAN"
+    },
+    "unit_commitment_integer": {
+      "UoM": null,
+      "constraints": null,
+      "default": false,
+      "description": "Whether the unit commitment variables are integer or not.",
+      "type": "BOOLEAN"
+    },
+    "unit_commitment_method": {
+      "UoM": null,
+      "constraints": {
+        "OneOf": [
+          null,
+          "basic"
```

Comment:
What is the intention here?  Does `null` mean no `unit_commitment`? In that case does TEM still expect this column in the tables?

I think the logically correct way to do this would be something like this:
```
{
  "unit_commitment_integer": {...},
  "unit_commitment_method": {
    ...
    "constraints": {
      "oneOf": ["basic", "something-else"]
    }
  },
  "required": ["unit_commitment_integer"]
}
```
This means, there will always be a `unit_commitment_integer` column, and `unit_commitment_method` will be present only when the previous one is true (which could be checked in code).

I understand if the current structure in TEM does not allow that, in that case this could be a future improvement.